fundingmanager.go

```diff
@@ -1435,20 +1584,191 @@ func (f *fundingManager) sendFundingLockedAndAnnounceChannel(
+
+	chainHash := *f.cfg.Wallet.Cfg.NetParams.GenesisHash
+
+	// Construct ChannelAnnouncement
+	chanAnn := &lnwire.ChannelAnnouncement{
+		ShortChannelID:	*shortChanID,
+		Features:	lnwire.NewFeatureVector([]lnwire.Feature{}),
+		ChainHash:	chainHash,
+	}
+
+	var chanFlags uint16
+
+	localKey := f.cfg.IDKey
+	remoteKey := completeChan.IdentityPub
+	selfBytes := localKey.SerializeCompressed()
+	remoteBytes := remoteKey.SerializeCompressed()
+	if bytes.Compare(selfBytes, remoteBytes) == -1 {
+		chanAnn.NodeID1 = localKey
+		chanAnn.NodeID2 = remoteKey
+		chanAnn.BitcoinKey1 = channel.LocalFundingKey
+		chanAnn.BitcoinKey2 = channel.RemoteFundingKey
+		chanFlags = 0
+	} else {
+		chanAnn.NodeID1 = remoteKey
+		chanAnn.NodeID2 = localKey
+		chanAnn.BitcoinKey1 = channel.RemoteFundingKey
+		chanAnn.BitcoinKey2 = channel.LocalFundingKey
+		chanFlags = 1
+	}
+
+	// Construct ChannelEdgeInfo from ChannelAnnouncement
+	var featureBuf bytes.Buffer
+	if err := chanAnn.Features.Encode(&featureBuf); err != nil {
+		fndgLog.Errorf("Unable to encode features into featureBuf: %v", err)
+		return err
+	}
+	edge := &channeldb.ChannelEdgeInfo{
+		ChannelID:	chanAnn.ShortChannelID.ToUint64(),
+		ChainHash:	chanAnn.ChainHash,
+		NodeKey1:	chanAnn.NodeID1,
+		NodeKey2:	chanAnn.NodeID2,
+		BitcoinKey1:	chanAnn.BitcoinKey1,
+		BitcoinKey2:	chanAnn.BitcoinKey2,
+		Features:	featureBuf.Bytes(),
+	}
+
+	// Add ChannelEdgeInfo to ChannelRouter's topology
+	err := f.cfg.AddEdge(edge); if err != nil {
+		fndgLog.Errorf("Unable to add edge to ChannelRouter's " +
+			"topology: %v", err)
+		return err
+	}
+
+	// Construct ChannelUpdate
+	chanUpdateAnn := &lnwire.ChannelUpdate{
+		ShortChannelID:	*shortChanID,
+		ChainHash:	chainHash,
+		Timestamp:	uint32(time.Now().Unix()),
+		Flags:		chanFlags,
+		TimeLockDelta:	uint16(f.cfg.DefaultRoutingPolicy.TimeLockDelta),
+		HtlcMinimumMsat:f.cfg.DefaultRoutingPolicy.MinHTLC,
+		BaseFee:	uint32(f.cfg.DefaultRoutingPolicy.BaseFee),
+		FeeRate:	uint32(f.cfg.DefaultRoutingPolicy.FeeRate),
+	}
+
+	// Generate signature
+	chanUpdateMsg, err := chanUpdateAnn.DataToSign()
+	if err != nil {
+		fndgLog.Errorf("Could not retrieve data to sign: %v", err)
+		return err
+	}
+	chanUpdateAnn.Signature, err = f.cfg.SignMessage(localKey, chanUpdateMsg)
+	if err != nil {
+		fndgLog.Errorf("Could not sign message: %v", err)
+		return err
+	}
+
+	// Construct ChannelEdgePolicy from ChannelUpdate
+	update := &channeldb.ChannelEdgePolicy{
+		Signature:	chanUpdateAnn.Signature,
+		ChannelID:	chanUpdateAnn.ShortChannelID.ToUint64(),
+		LastUpdate:	time.Unix(int64(chanUpdateAnn.Timestamp), 0),
+		Flags:		chanUpdateAnn.Flags,
+		TimeLockDelta:	chanUpdateAnn.TimeLockDelta,
+		MinHTLC:	chanUpdateAnn.HtlcMinimumMsat,
+		FeeBaseMSat:	lnwire.MilliSatoshi(chanUpdateAnn.BaseFee),
+		FeeProportionalMillionths:lnwire.MilliSatoshi(chanUpdateAnn.FeeRate),
+	}
+
+	// Add ChannelEdgePolicy to ChannelRouter's topology
+	err = f.cfg.UpdateEdge(update); if err != nil {
```

Comment:
I feel like it's preferable to use the normal `SendAnnouncement` code path to process the channel update. Only thing is that is is unclear whether the local node will send the channel_update to the remote peer since `AnnounceSignatures` is not sent, but it should do that regardless because it is not necessary to make a channel public.